### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: server/src/main/java/communication/LanguageVersion.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package communication;
+
+public class LanguageVersion {
```
Comment (EmiliaCiastek) [inline server/src/main/java/communication/LanguageVersion.java]: We should change name here

File: server/src/main/java/communication/MessageReceiver.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class MessageReceiver {
+    public String receiveMessage(BufferedReader reader) {
+        try {
+            return reader.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (EmiliaCiastek) [inline server/src/main/java/communication/MessageReceiver.java]: We should add exception handling

File: server/src/main/java/fleet/OneMastShip.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+class OneMastShip implements Ship {
+    private static final int NUMBER_OF_MASTS = 1;
+    private List<Integer> fields;
+
+    OneMastShip(List<Integer> positions) {
+        fields = new ArrayList<>();
```
Comment (rivalez) [inline server/src/main/java/fleet/OneMastShip.java]: you can use new Arraylist<>(positions)

File: server/src/main/java/communication/MessageReceiver.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package communication;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class MessageReceiver {
+    public String receiveMessage(BufferedReader reader) {
+        try {
+            return reader.readLine();
```
Comment (rivalez) [inline server/src/main/java/communication/MessageReceiver.java]: you can create Result string at the begging of the method. It is bad practice to have many return places

File: server/src/main/java/fleet/HardcodedFleet.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package fleet;
+
+import java.util.*;
+
+public class HardcodedFleet extends Fleet {
+
+    private static final Integer[] HARDCODED_POSITIONS = {0, 1, 2, 3, 17, 18, 19, 34, 40, 41, 44, 47, 54, 57, 61, 77, 78, 82, 85, 99};
```
Comment (rivalez) [inline server/src/main/java/fleet/HardcodedFleet.java]: why are you using Integer as objects?

File: server/src/test/java/communication/HitCheckerTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import fleet.HardcodedFleet;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class HitCheckerTest {
+
+    @Test
+    public void givenHardcodedFleetAndMastPositionWhenIsHitReturnHit() {
+        HitChecker hitChecker = new HitChecker(new HardcodedFleet());
```
Comment (rivalez) [inline server/src/test/java/communication/HitCheckerTest.java]: //given
//when
//then

File: server/src/test/java/communication/HitCheckerTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        int position = 0;
+
+        ShotState actualState = hitChecker.checkShot(position);
+        assertEquals(actualState, ShotState.HIT);
+    }
+
+    @Test
+    public void givenHardcodedFleetIncorrectMastPositionWhenIsHitReturnMissed(){
```
Comment (rivalez) [inline server/src/test/java/communication/HitCheckerTest.java]: hard to read this kind of methodName. maybe use _ or something

File: server/src/main/java/communication/ConnectionHandler.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    private void sendFireNotification(Socket socket, String messageToSend) {
+        MessageSender messageSender = new MessageSender();
+        PrintWriter printWriter = null;
+        try {
+            printWriter = new PrintWriter(socket.getOutputStream(), true);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        messageSender.send(printWriter, messageToSend);
```
Comment (igasm) [inline server/src/main/java/communication/ConnectionHandler.java]: so you can have printWriter as null and send it to messageSender.send(..) method

**Pull request merged**